discord/guild.py

```diff
@@ -764,6 +766,22 @@ def get_emoji(self, emoji_id: int, /) -> Optional[Emoji]:
         if emoji and emoji.guild == self:
             return emoji
         return None
+    
+    def get_guild_shop(self) -> Optional[GuildShop]:
+        """Returns the guild shop.
+
+        .. versionadded:: 2.4
+
+        Returns
+        -------
+        Optional[:class:`GuildShop`]
+            The returned Guild Shop or ``None`` if no shop is enabled.
+        """
+
+        shop = self._state.http.get_guild_shop(self.id)
```

Comment:
```suggestion
        shop = await self._state.http.get_guild_shop(self.id)
```
This is a HTTP request call, it needs to be awaited. Otherwise `shop`'s value will just be a coroutine.

Reply:
Awaited the http call (forgot it was a coroutine, that is why I didn't await it).